tools/dataset/preprocess_h36m.py

```diff
@@ -0,0 +1,347 @@
+            kps_2d_17 = np.array(cdf['Pose'])
+
+        num_frames = kps_2d_17.shape[1]
+        kps_2d_17 = kps_2d_17.reshape((num_frames, 32, 2))[::self.sample_rate,
+                                                           self.movable_joints]
+        kps_2d_17 = np.concatenate(
+            [kps_2d_17, np.ones((len(kps_2d_17), 17, 1))], axis=2)
+        kps_2d = np.zeros((len(kps_2d_17), 24, 3))
+        kps_2d[:, self.joints_mapping, :] = kps_2d_17[:, :, :]
+
+        # load 3D keypoints
+        with pycdf.CDF(
+                join(subj_dir, 'MyPoseFeatures', 'D3_Positions_mono',
+                     basename + '.cdf')) as cdf:
+            kps_3d_17 = np.array(cdf['Pose'])
+
+        kps_3d_17 = kps_3d_17.reshape(
+            (num_frames, 32, 3))[::self.sample_rate,
+                                 self.movable_joints] / 1000.
+        kps_3d_17 = np.concatenate(
+            [kps_3d_17, np.ones((len(kps_3d_17), 17, 1))], axis=2)
+        kps_3d = np.zeros((len(kps_3d_17), 24, 4))
+        kps_3d[:, self.joints_mapping, :] = kps_3d_17[:, :, :]
+
+        # calculate bounding boxes
+        bboxes = np.stack([
+            np.min(kps_2d_17[:, :, 0], axis=1),
+            np.min(kps_2d_17[:, :, 1], axis=1),
+            np.max(kps_2d_17[:, :, 0], axis=1),
+            np.max(kps_2d_17[:, :, 1], axis=1)
+        ],
+                          axis=1)
+        centers = np.stack([(bboxes[:, 0] + bboxes[:, 2]) / 2,
+                            (bboxes[:, 1] + bboxes[:, 3]) / 2],
+                           axis=1)
+        scales = self.scale_factor * np.max(
+            bboxes[:, 2:] - bboxes[:, :2], axis=1) / 200
+
+        # extract frames and save imgnames
+        imgnames = []
+        video_path = join(subj_dir, 'Videos', basename + '.mp4')
+        sub_base = subject + '_' + basename.replace(' ', '_')
+        img_dir = join(self.processed_dir, 'imgs', subject, sub_base)
+        os.makedirs(img_dir, exist_ok=True)
+        prefix = join(subject, sub_base, sub_base)
+
+        cap = cv2.VideoCapture(video_path)
+        for i in range(0, num_frames, self.sample_rate):
+            imgname = prefix + '_%06d.jpg' % (i + 1)
+            imgnames.append(imgname)
+            dest_path = join(self.processed_dir, 'imgs', imgname)
+            if os.path.exists(dest_path):
+                continue
+            cap.set(1, i)
+            success, img = cap.read()
+            if success:
+                cv2.imwrite(dest_path, img)
+        cap.release()
+        imgnames = np.array(imgnames)
+        assert len(centers) == len(imgnames)
+
+        print('Annoatations for sequence \"%s %s\" are loaded.'
+              '%d samples in total.' % (subject, basename, len(imgnames)))
+
+        return imgnames, centers, scales, kps_2d, kps_3d
+
+
+if __name__ == '__main__':
+    METADATA = '/data/Human3.6M/metadata.xml'
+    ORIGINAL_DIR = '/data/Human3.6M/original'
+    EXTRACTED_DIR = '/data/Human3.6M/extracted'
```

Comment:
EXTRACTED_DIR -> EXTRACT_DIR

The folder name does not have to be changed.

Reply:
extract and process in the same folder?

Reply:
nonono, ignore it

Reply:
I was uneasy with the variable name `EXTRACTED` because they are to be extracted and are not extracted yet... But nevermind